rust-symcrypt/src/gcm.rs

```diff
@@ -0,0 +1,409 @@
+/* Galois Counter Mode functions. For further documentation please refer to symcrypt.h */
+
+use crate::block_ciphers::*;
+use crate::errors::SymCryptError;
+use core::ffi::c_void;
+use std::mem;
+use std::ptr;
+use std::vec;
+use symcrypt_sys;
+
+pub struct GcmState {
+    expanded_key: symcrypt_sys::SYMCRYPT_GCM_EXPANDED_KEY,
+    state: symcrypt_sys::SYMCRYPT_GCM_STATE,
+}
+
+// Have to Box<> since memory address for Self is moved around when returning from GcmState::new()
+// My guess is that it has to do with Result<> wrapping the Self and SymCryptError.
+// The only accepted Cipher for GCM is AesBlock.
+impl GcmState {
+    pub fn new(
+        key: &[u8],
+        nonce: &[u8],
+        cipher: BlockCipherType,
+    ) -> Result<Box<Self>, SymCryptError> {
+        let mut instance = Box::new(GcmState {
+            state: symcrypt_sys::SYMCRYPT_GCM_STATE::default(),
+            expanded_key: symcrypt_sys::SYMCRYPT_GCM_EXPANDED_KEY::default(),
+        });
+
+        expand_key(key, &mut instance.expanded_key, convert_cipher(cipher))?;
+        unsafe {
+            // SAFETY: FFI calls
+            symcrypt_sys::SymCryptGcmInit(
+                &mut instance.state,
+                &instance.expanded_key,
+                nonce.as_ptr(),
+                nonce.len() as symcrypt_sys::SIZE_T,
+            );
+            Ok(instance)
+        }
+    }
+
+    pub fn auth(&mut self, auth_data: &[u8]) {
```

Comment:
```suggestion
    pub fn auth_part(&mut self, auth_data: &[u8]) {
```
to be consistent with others, and to indicate that this function can be called repeatedly

Reply:
Will change